OneCommunityGlobal/HighestGoodNetworkApp
# Mounica - Create a Weekly Summaries Reports recipient button for Owners

# Description
Show and editable list of who is currently receiving the Weekly summary report and
it's password protected with the test account 
[mounica_test20@gmail.com]
Password@1234

![image](https://github.com/OneCommunityGlobal/HGNRest/assets/49270410/e287dd82-e5a0-4920-99e7-5f7e145ad4f6)


## Related PRS (if any):
To test this frontend PR you need to checkout the #566 backend PR.
https://github.com/OneCommunityGlobal/HGNRest/pull/566

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as admin user
4. go to Reports→ Weekly Summaries Reports →  Weekly Summaries Reports button
5. Enter the password [Password@1234]  and if the user is Authorized, a popup shows up, with list of Weekly Summary recipients
6. Test if you're able to add/delete the Weekly Summary recipients.
7. The weekly summaries will be sent to the recipients through email. In order to test this, add yourself to the recipients and to send/receive email using Gmail API, using OAuth2 and nodemailer, kindly follow the below video and setup up a test mail id.
https://www.youtube.com/watch?v=-rcRf7yswfM&t=10s&ab_channel=MafiaCodes

Fill below variable values from the above setup in the .env file of the backend.
REACT_APP_EMAIL =
REACT_APP_EMAIL_CLIENT_ID =
REACT_APP_EMAIL_CLIENT_SECRET =
REACT_APP_EMAIL_REFRESH_TOKEN =
REACT_APP_EMAIL_CLIENT_REDIRECT_URI = https://developers.google.com/oauthplayground
sendEmail = TRUE

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HGNRest/assets/49270410/19ef1602-fa6c-4ea2-b923-b78b2527735c


![Screenshot from 2023-10-10 13-57-37](https://github.com/OneCommunityGlobal/HGNRest/assets/49270410/3ea7cc36-ca8f-415d-87ee-4a83a888be03)


## Note:
Include the information the reviewers need to know.
